### DISCUSSION TITLE
Traffic encryption

### DISCUSSION THREAD

**Discussion by aamironov** [containers/netavark#674] [General]
Docker has option to encrypt the traffic in docker network. Is there any available plugin for netavark to apply overlay encryption for the podman network?

**Comment by Luap99** [comment]
Podman/Netavark is single node only, do you want to encrypt network traffic on the same host?
It sounds to me docker only uses this overlay networks to connect different hosts in a docker swarm setup.

**Reply by aamironov** [reply]
Yes, I am looking for traffic encryption on the same host and it looks for me as I need some plugin for netavark to do that, but I wasn't able to find something useful for it. May be I just need to implement one.
Overlay networks are usually used for docker swarms and for Podman pods.

**Reply by Luap99** [reply]
What do you gain by encrypting traffic on the same host? You still have the unencrypted data anyway on this host.
Netavark does not support any multi host (overlay) networks so I really do not understand the use case.

Netavark plugins were just released with v1.6 so you could write your own stuff if you really have to.
There is some WIP wireguard implementation which could be used to do this. https://github.com/containers/netavark/pull/472
